### ISSUE TITLE
caretRender needs more than just direction

### ISSUE THREAD

**Issue by innopal** [AllenFang/react-bootstrap-table#740]
It would be really nice to add row and column parameters besides direction to caretRender, similar to the signature that sortFunc has.

I need to initially set the caret direction based on the last sort that user did to their table.
So for example if the list was sorted by name field with desc direction last time user sorted the list, I need to preserve that when they come back.

Cheers

**Comment by AllenFang** [comment]
> for example if the list was sorted by name field with desc direction last time user sorted the list, I need to preserve that when they come back.

I've no idea about this, is that you hope ```react-bootstrap-table``` to keep the last sort condition and restore it when they back?

and I dont know what's means for **when they come back**, any more description.

Thanks :)

**Comment by innopal** [comment]
I'd be glad to provide more description.

Never mind the 'user coming back part'.
What I like to do is for all the column in the table I'd like to hide the sort caret until user hovers over the column header then I display the caret.
That is except for the column that is currently sorted. For the currently sorted column I'd like to display the caret with or without hovering.
For that in my custom caretRender function I'd like to be able to identify the caret I am rendering belongs to which column, but I am unable to do that currently because caretRender currently only accepts a single parameter (direction).
I'd like it to accept 3 parameters (cell, row and direction) so that I can distinguish between when I am rendering the caret for the currently sorted column or for every other columns.

Hope that is more clear now.

**Comment by innopal** [comment]
@AllenFang , for now just having cell parameter (dataField) is sufficient btw.
I thought though having the entire row will be useful in the future.

Thanks

**Comment by innopal** [comment]
I looked at the code. This could be easily done by changing:

```
if (caretRender) {
      sortCaret = caretRender(sort);
    }
```

to 

```
if (caretRender) {
      sortCaret = caretRender(sort, this.props.dataField);
    }

```

**Comment by AllenFang** [comment]
@innopal, ok, I can pass the ```dataField``` as the second argument in next version :)

**Comment by AllenFang** [comment]
Fixed on ```v2.5.5```.
